policy/lib/purl.rego

```diff
@@ -0,0 +1,26 @@
+package lib.purl
+
+import rego.v1
+
+# Extract the repoid
+repoid(purl) := _named_qualifier("repository_id", purl)
```

Comment:
I know it sounds a bit counter-intuitive, but I find that in rego an abundance of helper methods makes the code harder to maintain, especially when they are spread out across different packages. The tooling for navigating the code base is just not as sophisticated as the tooling for other languages like Go, python, etc.

I would suggest writing the policy rules first, using local helpers (same file) where needed, and only then decide if anything should be abstracted out into a lib package.

Reply:
Yeah, that's reasonable. This code could probably live in the package where the denies are defined. I'll leave it in draft while considering that.